### PR TITLE
Jse dev

### PR DESCRIPTION
**Pull request by cjwest**
mkay, this is ready for testing!

Please run both dev and deploy suites for a newly deployed JSE site.
behat -p _profile_of_newly_deployed_jse_site_ --suite deploy features
behat -p _profile_of_newly_deployed_jse_site_ --suite dev features

If you want to put it in a file and check it out later:
behat -p _profile_of_newly_deployed_jse_site_ --suite dev -o _your_file_name.out_ features&
tail -f _your_file_name.out_

### CODE REVIEW

**Comment by cjwest** [comment]
@sherakama would you like to review this?

**Comment by cjwest** [comment]
@sherakama, @boznik, and @jbickar:

I just added a draft of my "Using Behat" at: 
https://github.com/SU-SWS/linky_clicky/blob/jse-dev/UsingLinkClicky.md

Before I take the time to dot the "i"s and cross the "t"s, I'd really like it if you could give me feedback:
- What sections should I keep/remove?
- What needs more help?

Really appreciate it!

Caryl

**Comment by jbickar** [comment]
Looks good. You start to trail off and get a little folksy and "note to self" there at the end, though :)

Who is the target audience? Remember that this will be public-facing documentation.

File: includes/features/SU-SWS/stanford_event_series/stanford_event_series.feature [hunk 1]
```diff
@@ -9,46 +9,6 @@ Feature: Stanford Event Series
     When I go to "node/add/stanford-event-series"
     Then I should see "Access denied"
 
-  @api @dev @destructive
-  Scenario: Administrator can create event series nodes with events attached
```
Comment (sherakama) [inline includes/features/SU-SWS/stanford_event_series/stanford_event_series.feature:13]: This test seems like a good test to have here. Why did you remove it?

File: includes/features/SU-SWS/stanford_event_series/stanford_event_series.feature [hunk 2]
```diff
@@ -9,46 +9,6 @@ Feature: Stanford Event Series
...
-    Then I click on the element with css selector ".views-row-first .views-field-title a"
-    Then I click "Edit" in the "Tabs" region
-    Then I check "Foo"
-    Then I press "Save"
-    Then I should see "This event belongs to the following series"
-    Then I go to "events/series/foo"
-    Then I should see 1 or more ".views-row" elements
-
-  @api @dev @destructive
-  Scenario: Site owner can create event series types
```
Comment (sherakama) [inline includes/features/SU-SWS/stanford_event_series/stanford_event_series.feature:33]: Agreed. We don't need to test for the site owner in this shared feature.

File: includes/features/SU-SWS/stanford_event_series/stanford_event_series.feature [hunk 3]
```diff
@@ -70,14 +30,18 @@ Feature: Stanford Event Series
...
+  Scenario: Administrator can create event series nodes with events attached
+    Given I am logged in as a user with the "site owner" role
+    When I go to "node/add/stanford-event-series"
+    Then I should see "Create Stanford Event Series" in the "Branding" region
+    Then I fill in "edit-title" with "Foo"
+    Then I attach the file "img/ooooaaaahhh.jpg" to "edit-field-s-image-info-und-0-field-s-image-image-und-0-upload"
+    Then I press "Upload"
+    Then I press "Save"
+    Then I should see 1 or more ".field-name-field-s-image-image" elements
+    Then I go to "admin/manage/events"
```
Comment (sherakama) [inline includes/features/SU-SWS/stanford_event_series/stanford_event_series.feature]: This step doesn't appear to belong in this scenario.

File: includes/features/SU-SWS/stanford_event_series/stanford_event_series.feature [hunk 4]
```diff
@@ -70,14 +30,18 @@ Feature: Stanford Event Series
...
-    #Then I click "Edit" in the "Tabs" region
-    #Then I check "Foo"
-    #Then I press "Save"
-    #Then I should see "This event belongs to the following series"
-    #Then I go to "events/series/foo"
-    #Then I should see 1 or more ".views-row" elements
+
+  @api @dev @destructive
+  Scenario: Administrator can create event series nodes with events attached
+    Given I am logged in as a user with the "site owner" role
```
Comment (sherakama) [inline includes/features/SU-SWS/stanford_event_series/stanford_event_series.feature]: We should not use the site owner role in the share features.

File: includes/features/SU-SWS/stanford_event_series/stanford_event_series.feature [hunk 5]
```diff
@@ -70,14 +30,18 @@ Feature: Stanford Event Series
...
-    #Then I click on the element with css selector ".views-row-first .views-field-title a"
-    #Then I click "Edit" in the "Tabs" region
-    #Then I check "Foo"
-    #Then I press "Save"
-    #Then I should see "This event belongs to the following series"
-    #Then I go to "events/series/foo"
-    #Then I should see 1 or more ".views-row" elements
+
+  @api @dev @destructive
+  Scenario: Administrator can create event series nodes with events attached
```
Comment (sherakama) [inline includes/features/SU-SWS/stanford_event_series/stanford_event_series.feature]: Scenario title doesn't match tests

File: includes/features/SU-SWS/stanford_event_series/stanford_event_series.feature [hunk 6]
```diff
@@ -70,14 +30,18 @@ Feature: Stanford Event Series
     Then I attach the file "img/ooooaaaahhh.jpg" to "edit-field-s-image-info-und-0-field-s-image-image-und-0-upload"
     Then I press "Upload"
     Then I press "Save"
-    #todo: Then I should be on "events/series/foo"
     Then I should see 1 or more ".field-name-field-s-image-image" elements
     Then I go to "admin/manage/events"
```
Comment (sherakama) [inline includes/features/SU-SWS/stanford_event_series/stanford_event_series.feature:22]: This step doesn't add value to this scenario imho.

File: includes/features/SU-SWS/stanford_person/stanford_person.feature [hunk 7]
```diff
@@ -3,7 +3,7 @@ Feature: Stanford Person
   As an end user and administrator
   I want to check for the creation and display of people content.
 
-  @api @javascript @contrib @stanford @destructive
+  @api @dev @destructive
   Scenario: Stanford Person
     Given the "stanford_person" module is enabled
```
Comment (sherakama) [inline includes/features/SU-SWS/stanford_person/stanford_person.feature]: Please remove this line. We don't want to be enabling anything.

File: includes/features/SU-SWS/stanford_person/stanford_person.feature [hunk 8]
```diff
@@ -3,7 +3,7 @@ Feature: Stanford Person
...
     And the "entity_token" module is enabled
```
Comment (sherakama) [inline includes/features/SU-SWS/stanford_person/stanford_person.feature]: Same here.

File: includes/features/SU-SWS/stanford_person/stanford_person.feature [hunk 9]
```diff
@@ -3,7 +3,7 @@ Feature: Stanford Person
   As an end user and administrator
   I want to check for the creation and display of people content.
 
-  @api @javascript @contrib @stanford @destructive
+  @api @dev @destructive
   Scenario: Stanford Person
```
Comment (sherakama) [inline includes/features/SU-SWS/stanford_person/stanford_person.feature]: This should be more descriptive.

File: includes/features/SU-SWS/stanford_person/stanford_person.feature [hunk 10]
```diff
@@ -17,239 +17,13 @@ Feature: Stanford Person
...
-
-  @api @prod @dev @deploy @stanford @live
-  Scenario: Affiliates column on Manage Person
-    Given I am logged in as a user with the "administrator" role
-    When I go to "admin/manage/people"
-    Then I should see 1 or more ".views-field-field-s-person-affiliation" elements
+  @deploy
+  Scenario: Stanford Person CAP node page
+    Given I am on "people/jacob-smith"
+    Then I should see a ".postcard-left" element
```
Comment (sherakama) [inline includes/features/SU-SWS/stanford_person/stanford_person.feature]: Why are we testing for .postcard-left? If we are testing to see if the person page exists the test for the url is fine. If you are testing to see if the content and other elements are available this test should be added to.

File: includes/features/SU-SWS/stanford_person/stanford_person_cap_node_display.feature [hunk 11]
```diff
@@ -3,62 +3,28 @@ Feature: Stanford CAP Person Node Display
...
-
-  @content @deploy @stanford
-  Scenario: Faculty node
-    Given I am on "people/faculty"
-    And I click "Jacob Smith" in the "Content Body" region
-    Then I should see "Professor of English" in the "Content Body" region
-
-  @content @live @deploy @stanford @safe @contrib @stanford
+  @api @dev @destructive
+  Scenario: Stanford Person
```
Comment (sherakama) [inline includes/features/SU-SWS/stanford_person/stanford_person_cap_node_display.feature]: The scenario should be more descriptive.

File: includes/features/SU-SWS/stanford_person/stanford_person_cap_node_display.feature [hunk 12]
```diff
@@ -3,62 +3,28 @@ Feature: Stanford CAP Person Node Display
...
+    Given the "stanford_person" module is enabled
```
Comment (sherakama) [inline includes/features/SU-SWS/stanford_person/stanford_person_cap_node_display.feature]: Please, no module is enabled.

File: includes/features/SU-SWS/stanford_person/stanford_person_cap_node_display.feature [hunk 13]
```diff
@@ -3,62 +3,28 @@ Feature: Stanford CAP Person Node Display
...
+    And I should see "Person Leland Stanford has been created"
+
+  @api @live
+  Scenario: Affiliates column on Manage Person
+    Given I am logged in as a user with the "administrator" role
+    When I go to "admin/manage/people"
+    Then I should see 1 or more ".views-field-field-s-person-affiliation" elements
+
+  @deploy
   Scenario: Stanford Person CAP node page
```
Comment (sherakama) [inline includes/features/SU-SWS/stanford_person/stanford_person_cap_node_display.feature:25]: This test is a duplicate of the stanford_person.feature test. Please remove.

File: includes/features/SU-SWS/stanford_private_page/stanford_private_page.feature [hunk 14]
```diff
@@ -3,16 +3,6 @@ Feature: Stanford Private Page
...
-    When I wait for the Site Actions drop down to appear
-    And I click "Add Private Page" in the "Admin Shortcuts" region
-    Then I should see "Create Private Page" in the "Branding" region
-    Then I fill in "edit-title" with "Private Page Test"
-    And I press the "Save" button
-    Then I should see "Private Page Private Page Test has been created" in the "Console" region
-
   @api @dev @destructive
   Scenario: Site owner can create and edit Private Page types
     Given I am logged in as a user with the "site owner" role
```
Comment (sherakama) [inline includes/features/SU-SWS/stanford_private_page/stanford_private_page.feature]: We should use the administrator role when testing creation of content types in the shared directory.

File: includes/features/SU-SWS/stanford_private_page/stanford_private_page.feature [hunk 15]
```diff
@@ -30,9 +20,8 @@ Feature: Stanford Private Page
     Then I go to "private/bar"
```
Comment (sherakama) [inline includes/features/SU-SWS/stanford_private_page/stanford_private_page.feature]: This step is not needed.

File: includes/features/SU-SWS/stanford_private_page/stanford_private_page.feature [hunk 16]
```diff
@@ -30,9 +20,8 @@ Feature: Stanford Private Page
...
     And I should see "Bar" in the "Content Head" region
 
-  @api @javascript @safe
+  @api @safe
   Scenario: As a site owner I can see Create Private Page
```
Comment (sherakama) [inline includes/features/SU-SWS/stanford_private_page/stanford_private_page.feature]: We should not be using the site owner role for testing here. Change to administrator.

File: README.md [hunk 17]
```diff
@@ -90,49 +90,6 @@ Directory structure
...
-* deploy: Runs `@deploy`, but not `@dev`, `@destructive`, `@content`, or `@site`
-* content: Runs only `@content`
-* site: Runs only `@site`
-* contrib: Runs `@contrib`, but not `@dev` or `@destructive`
-* contrib-dev: Runs all `@contrib`
-* stanford: Runs all `@stanford`, but not `@dev` or `@destructive`
-
-## Running suites and profiles
-You can run a suite and a profile at the same time. eg: `behat --profile local --suite live`
+[Using Linky Clicky] (https://github.com/SU-SWS/linky_clicky/blob/jse-dev/README.md)
```
Comment (sherakama) [inline README.md]: This is not linking to the correct place.

File: README.md [hunk 18]
```diff
@@ -90,49 +90,6 @@ Directory structure
...
+------------------
+See [Using Linky Clicky] (https://github.com/SU-SWS/linky_clicky/blob/jse-dev/README.md)
```
Comment (sherakama) [inline README.md]: This is not linking to the correct place.

File: UsingLinkClicky.md [hunk 19]
```diff
@@ -0,0 +1,396 @@
...
+4. Hosting Environment: Where is this site hosted? Is it local on your machine? Is it on sites? On Anchorage? 
+
+5. Authentication Scheme: What approach do use to authenticate? Native Drupal? Webauth? SamL?
+
+### Lifecycle
+
+Testing for our product development and subsequent sites has a sort of timeline or lifecycle. Which looks roughly like this:
+
+    Requirements
+        > Development
```
Comment (sherakama) [inline UsingLinkClicky.md]: Don't we deploy a website before we develop on it?

Suggested update:

 /- Reqs
 /-- Deploy
 /--- Dev
 /---- Production
 /----- Launch
 /------ Live

File: UsingLinkClicky.md [hunk 20]
```diff
@@ -0,0 +1,396 @@
...
+
+In the requirements phase we define the functionality. This functionality might be for a new or updated product, theme, behavior, content, etc. Ideally, this is the phase in which we identify and draft our first Behat tests. 
+
+#### Development
+
+During development we are adding to the functionality of the product, so here’s where we thoroughly test functionality. Since you can expect that the site will get rebuilt, anything goes. In our tests, we can add random content, delete existing pages, change the name of the site, etc.
+
+#### Deployment
+
+The deployment phase signals that we are no longer in development. When we deploy a site for a customer, we need to assume that we fully tested product functionality in development. In this phase we want to ensure that the site was deployed properly. We need to verify that the modules, themes, and content were all enabled and imported properly. We want happy customers!
```
Comment (sherakama) [inline UsingLinkClicky.md]: This is not always true. We don't always deploy a fully functional website. Most of the time we deploy a website and then iterate on.

File: UsingLinkClicky.md [hunk 21]
```diff
@@ -0,0 +1,396 @@
...
+
+#### Customization
+
+For some of our clients we build extensive and unique functionality for their site. Although development occurs during this phase, it often occurs in tandem with the production phase on a site and we cannot expect the site to be rebuilt. Testing in the phases verifies functionality yet cannot change anything on the site.
```
Comment (sherakama) [inline UsingLinkClicky.md]: This is synonymous with development IMHO.

File: UsingLinkClicky.md [hunk 22]
```diff
@@ -0,0 +1,396 @@
...
+  </tr>
+  <tr>
+    <td>dev</td>
+    <td>Everything but >></td>
+    <td>@live @launch</td>
+  </tr>
+  <tr>
+    <td>deploy</td>
+    <td>@safe @deploy </td>
+    <td>@destructive @dev @launch @live</td>
```
Comment (sherakama) [inline UsingLinkClicky.md]: Do we need to explicitly exclude live? If so we should explicitly exclude launch as well.

File: UsingLinkClicky.md [hunk 23]
```diff
@@ -0,0 +1,396 @@
...
+  </tr>
+  <tr>
+    <td>launch</td>
+    <td>@safe @launch </td>
+    <td>@destructive @dev @deploy @live </td>
```
Comment (sherakama) [inline UsingLinkClicky.md]: Do we need to explicitly exclude live? If so we should explicitly exclude deploy as well.

File: UsingLinkClicky.md [hunk 24]
```diff
@@ -0,0 +1,396 @@
...
+  </tr>
+  <tr>
+    <td>live</td>
+    <td>@safe @live @launch </td>
+    <td>@destructive @dev @deploy </td>
```
Comment (sherakama) [inline UsingLinkClicky.md]: Do we need to also exclude launch and deploy?

**Comment by sherakama** [comment]
Lots of inline comments above but just a general note about UsingLinkyClicky.md; I agree with John about the "folksy" comments at the end of the document. It seems like you have a number of questions. Maybe move them to a "Todo" section or create tickets for them. I think this document should reflect what the status of the testing is at this moment.

File: includes/features/SU-SWS/stanford_person/stanford_person_cap_node_display.feature (hunk 13 above)
Comment (cjwest) [inline includes/features/SU-SWS/stanford_person/stanford_person_cap_node_display.feature:25]: Actually, because the cap node display is different from the default node display, and yet it is part of the feature, we need to separate these tests. They are mutually exclusive.

**Comment by cjwest** [comment]
Hi @sherakama ,
I addressed your comments and moved out a bunch of site owner and editor tests.

**Comment by cjwest** [comment]
@sherakama , I don't want this to grow stale, so here's a ping.

File: includes/features/SU-SWS/stanford_person/stanford_person_cap_node_display.feature (hunk 13 above)
Comment (sherakama) [inline includes/features/SU-SWS/stanford_person/stanford_person_cap_node_display.feature:25]: So one would run one of the feature files and not the other?

Comment (sherakama) [inline includes/features/SU-SWS/stanford_person/stanford_person_cap_node_display.feature:25]: I don't want this to be a blocker for this test. I'll merge and then we can carry on with new issues.

File: products/jse/features/jse_editor.feature [hunk 25]
```diff
@@ -3,9 +3,11 @@ Feature: Editor
   As an Editor
   I want to check for my ability to complete my tasks
 
+  Background:
```
Comment (sherakama) [inline products/jse/features/jse_editor.feature:7]: This background would require all scenarios to use the @api tag in this feature. Take the tag of of each individual scenario and place the @api tag as the very first line in this file.

File: products/jse/features/jse_site_owner.feature [hunk 26]
```diff
@@ -3,22 +3,22 @@ Feature: Site Owner
   As a Site Owner
   I want to check for my ability to complete my tasks
 
+  Background:
```
Comment (sherakama) [inline products/jse/features/jse_site_owner.feature:8]: This background would require that all scenarios use the @api tag. To avoid hilarity later down the road please remove all @api tags from each scenario and place one @api tag at the very first line of this file.

**Comment by cjwest** [comment]
@sherakama , thanks for that feedback! I updated both site owner and editor features.

**Comment by tsui1** [comment]
:tada:
